workitem/expression_compiler.go

```diff
@@ -124,6 +128,24 @@ func (c *expressionCompiler) Equals(e *criteria.EqualsExpression) interface{} {
 	return c.binary(e, "=")
 }
 
+func (c *expressionCompiler) Substring(e *criteria.SubstringExpression) interface{} {
+	if isInJSONContext(e.Left()) {
+		l := e.Left().(*criteria.FieldExpression).FieldName
```

Comment:
In order to avoid a panic, you should do the conversion to `*criteria.FieldExpression` in a separate step and see if it worked (using the second `bool` return parameter).

Reply:
Fixed. 675da0ea05afa2576338e2d44c3dcbb916bd3f63